src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c

```diff
@@ -62,9 +62,11 @@ static int
 xocl_client_lock_bitstream(struct xocl_dev *xdev, struct client_ctx *client)
 {
 	int ret = 0;
-	mutex_lock(&client->lock);
+	mutex_lock(&client->lock);         // protect current client
+	mutex_lock(&xdev->ctx_list_lock);  // protect xdev->xclbin_id
 	ret = xocl_client_lock_bitstream_nolock(xdev,client);
 	mutex_unlock(&client->lock);
+	mutex_unlock(&xdev->ctx_list_lock);
```

Comment:
mutex unlock should be done in the reverse order of locking for nested mutex.

Reply:
Yes, of course; my bad.